code/datums/status_effects/neutral.dm

```diff
@@ -135,3 +135,126 @@
+	var/mob/living/L = usr
+	if(L != owner)
+		return
+	QDEL_NULL(attached_effect)
+	qdel(src)
+
+/obj/screen/alert/status_effect/protector/Click()
+	var/mob/living/L = usr
+	if(L != owner)
+		return
+	QDEL_NULL(attached_effect)
+	qdel(src)
+
+/datum/status_effect/aegis/on_creation(mob/living/new_owner, mob/living/protector)
+	src.protector = protector
+	linked_alert_protector = protector.throw_alert(id, alert_type_protector)
+	linked_alert_protector.attached_effect = src
+	RegisterSignal(new_owner, COMSIG_CARBON_ATTACKED_BY, .proc/interrupt_hit)
+	RegisterSignal(new_owner, COMSIG_MOVABLE_MOVED, .proc/check_protectee_dist)
+	RegisterSignal(new_owner, list(COMSIG_PROJECTILE_PREHIT, COMSIG_PROJECTILE_POINT_BLANK), .proc/interrupt_bullet)
+	RegisterSignal(protector, COMSIG_MOVABLE_MOVED, .proc/take_me_with_you)
+	RegisterSignal(protector, SIGNAL_ADDTRAIT(TRAIT_INCAPACITATED), .proc/cancel_protection)
+	..()
+
+/datum/status_effect/aegis/on_apply()
+	owner.visible_message("<span class='notice'><b>[protector] covers [owner] with [protector.p_their()] body, guarding against attacks!</b></span>", "<span class='notice'><b>[protector] covers you with [protector.p_their()] body, guarding against attacks!</b></span>")
+	examine_text = "<span class='notice'><b>[owner.p_they(TRUE)] [owner.p_are()] being guarded closely by [protector].</b></span>"
+	original_alpha = owner.alpha
+	owner.alpha = 100
+	playsound(owner, 'sound/weapons/fwoosh.ogg', 75, FALSE)
+	protector.forceMove(owner.loc)
+	owner.SetKnockdown(duration)
+	owner.SetImmobilized(paralyze_duration)
+	apply_defensive_buff()
+	return ..()
+
+/datum/status_effect/aegis/on_remove()
+	if(owner && protector)
+		owner.visible_message("<span class='notice'>[protector] ceases guarding [owner].</span>", "<span class='notice'><b>[protector] ceases guarding you.</b></span>")
+
+	if(owner)
+		owner.alpha = original_alpha
+		UnregisterSignal(owner, list(COMSIG_CARBON_ATTACKED_BY, COMSIG_MOVABLE_MOVED, COMSIG_PROJECTILE_PREHIT, COMSIG_PROJECTILE_POINT_BLANK))
+		owner.SetKnockdown(0)
+		owner.SetImmobilized(0)
+
+	if(protector)
+		protector.clear_alert(id)
+		remove_defensive_buff()
+		UnregisterSignal(protector, list(COMSIG_MOVABLE_MOVED, SIGNAL_ADDTRAIT(TRAIT_INCAPACITATED)))
+
+/datum/status_effect/aegis/proc/apply_defensive_buff()
+	protector.SetKnockdown(0)
+	protector.add_movespeed_modifier(/datum/movespeed_modifier/mob_protector)
+	if(ishuman(protector))
+		var/mob/living/carbon/human/H = protector
+		H.physiology.brute_mod *= defensive_buff_factor
+		H.physiology.burn_mod *= defensive_buff_factor
+		H.physiology.stamina_mod *= defensive_buff_factor
+
+/datum/status_effect/aegis/proc/remove_defensive_buff()
+	protector.remove_movespeed_modifier(/datum/movespeed_modifier/mob_protector)
+	if(ishuman(protector))
+		if(owner)
+			protector.start_pulling(owner, GRAB_PASSIVE, supress_message=TRUE)
+		var/mob/living/carbon/human/H = protector
+		H.physiology.brute_mod /= defensive_buff_factor
```

Comment:
what if you've changed races since applying the aegis?

Reply:
then that sucks i guess, without a datumized damage mod system you're either going to short people who change races mid aegis, or people who get some other kind of damage mod mid aegis, so I'm sticking with what other parts of the code do which is the former